compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -1199,7 +1199,10 @@ object Denotations {
             val alt =
               if (generateStubs) missingHook(owner.symbol.moduleClass, selector)
               else NoSymbol
-            if (alt.exists) alt.denot else MissingRef(owner, selector)
+            if (alt.exists) alt.denot
+            else if (owner.symbol == defn.RootClass && (selector == nme.scala_ || selector == nme.scalaShadowing))
```

Comment:
Maybe document what this test or extract it to an inner function.

I don't have a good understanding of this does though. @smarter?